src/directive/sort-by.directive.ts

```diff
@@ -53,11 +65,23 @@ export class JhiSortByDirective implements AfterViewInit {
     }
 
     private applyClass() {
-        const childSpan = this.el.nativeElement.children[1];
-        let add = this.sortAscIcon;
-        if (!this.jhiSort.ascending) {
-            add = this.sortDescIcon;
+        let icon: any = this.sortIcon;
+        let iconName = this.sortIconName;
+        if (this.jhiSort.predicate === this.jhiSortBy) {
+            if (this.jhiSort.ascending) {
+                icon = this.sortAscIcon;
+                iconName = this.sortAscIconName;
+            } else {
+                icon = this.sortDescIcon;
+                iconName = this.sortDescIconName;
+            }
         }
-        this.renderer.setElementClass(childSpan, add, true);
-    };
+
+        const childIcon = this.el.nativeElement.children[1];
```

Comment:
`this.el.nativeElement.children[1].children[0]` may be?

Reply:
The label is the first element but it’s better to use the css selector « fa-icon »

Reply:
Or maybe « fa-icon p »